community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/PropertyContainerLocker.java

```diff
@@ -69,6 +69,34 @@ void release( Statement statement, ResourceType type, long resourceId )
         }
     }
 
+    /**
+     * The Cypher runtime keeps statements open for longer, so this method does not close the statement after itself
+     */
+    public Lock exclusiveLock( Statement statement, PropertyContainer container )
+    {
+        if ( container instanceof Node )
```

Comment:
BTW @systay you are claiming that the fix is needed in 2.3, but the PR is issued against 3.0 and it is also labelled as such, mistake?